helm/templates/daemonset.yaml

```diff
@@ -28,36 +30,40 @@ spec:
             - {{ $k }}.efs.{{ $v.region }}.amazonaws.com
       {{- end }}
     {{- end }}
-    {{- if .Values.imagePullSecrets }}
+    {{- with .Values.imagePullSecrets }}
       imagePullSecrets:
-      {{- range .Values.imagePullSecrets }}
-        - name: {{ . }}
-      {{- end }}
+        {{- toYaml . | nindent 8 }}
     {{- end }}
+      serviceAccountName: {{ include "aws-efs-csi-driver.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+    {{- with .Values.nodeSelector }}
       nodeSelector:
-        beta.kubernetes.io/os: linux
-        {{- with .Values.nodeSelector }}
-        {{- . | toYaml | nindent 8 }}
-        {{- end }}
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
       hostNetwork: true
-      {{- if .Values.dnsPolicy }}
+      priorityClassName: {{ .Values.priorityClassName }}
+    {{- if .Values.dnsPolicy }}
       dnsPolicy: "{{ .Values.dnsPolicy }}"
-      {{- end }}
-      {{- with .Values.dnsConfig }}
+    {{- end }}
+    {{- with .Values.dnsConfig }}
       dnsConfig:
         {{- toYaml . | nindent 8 }}
-      {{- end }}
-      priorityClassName: system-node-critical
+    {{- end }}
+    {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.tolerations }}
       tolerations:
-        - operator: Exists
```

Comment:
Sorry, I didn't notice this before but with this being a daemonset it should probably have a default set of tolerations that will work in many cases.  This toleration is one that tolerates any taint that exists.  In the ebs project they kept this as a default with a boolean value to turn it on an off.

Reply:
I see the relevant PRs for that. I give way too much credit to the admin for how they want to configure their tolerations. I can move `- operator: Exists` into values.yaml as the default for the chart, then the admin can configure it how they want later if they need to. This should keep the default behavior without the need of a bool.